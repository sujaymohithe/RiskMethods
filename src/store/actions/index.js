export {
    auth,
    logout,
    setAuthRedirectPath,
    authCheckState
} from './auth';

export {
    InitIndicatorMessages,
    InitIndicatorMessageDetails
} from './indicatorMessage';