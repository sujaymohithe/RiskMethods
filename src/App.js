import React, { useEffect } from 'react';
import './App.css';
import Header from './components/Header/Header';
import MainRouter from './MainRouter';
import * as actions from './store/actions/index';
import { connect } from 'react-redux';

const App = props => {
  const { onTryAutoSignup } = props;
  useEffect(() => {
    onTryAutoSignup();
  }, [onTryAutoSignup]);


  return (
    <div>
      <Header {...props} />
      <MainRouter {...props} />
    </div>
  );
}

const mapStateToProps = state => {
  return {
    isAuthenticated: state.auth.token !== null
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onTryAutoSignup: () => dispatch(actions.authCheckState())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
