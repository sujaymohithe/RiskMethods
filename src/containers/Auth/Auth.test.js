import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import App from '../../App';
import { store } from '../../store/store';
import AuthContainer from './Auth';

Enzyme.configure({ adapter: new Adapter() });


describe('<Auth/> Container', () => {
    const wrapper = shallow(
        <Provider store={store}>
            <AuthContainer />
        </Provider>
    );

    it('should render the AuthContainer', () => {
        expect(wrapper.length).toEqual(1);
    });

    const container = mount(
        <Provider store={store}>
            <AuthContainer />
        </Provider>
    );
    it('should have 2 input fields', () => {
        expect(container.find('input').length).toEqual(2);
    });
    
    it('renders the Auth form submit button successfully', () => {
        expect(container.find('button').length).toEqual(1);
    })

    it('should set the password value on change event', () => {
        container.find('input[type="password"]').simulate('change', {
            target: {
                value: 'somenewpassword',
            },
        });
        expect(container.find('input[type="password"]').prop('value')).toEqual(
            'somenewpassword',
        );
    });

    it('should have proper props for email field', () => {
        expect(container.find('input[type="username"]').prop('placeholder')).toEqual(
            'Enter your email address'
        );
    });

    it('test button click event', () => {
        const mockCallBack = jest.fn();
        const button = shallow((<input type="button" onClick={mockCallBack}>Ok!</input>));
        button.simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })
})