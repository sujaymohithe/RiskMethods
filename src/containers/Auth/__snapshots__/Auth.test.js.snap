// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth/> Container should match the snapshot 1`] = `"<div class=\\"Auth\\"><div class=\\"col-sm-3 ml-auto mr-auto pt-5\\"><form><div class=\\"form-group\\"><label class=\\"form-label\\" for=\\"username\\"><b>Email address</b></label><input type=\\"username\\" autofocus=\\"\\" value=\\"\\" name=\\"username\\" id=\\"username\\" class=\\"form-control\\"/></div><div class=\\"form-group\\"><label class=\\"form-label\\" for=\\"password\\"><b>Password</b></label><input type=\\"password\\" value=\\"\\" name=\\"password\\" id=\\"password\\" class=\\"form-control\\"/></div><button type=\\"submit\\" class=\\"col-sm-6 offset-sm-4 mt-4 btn btn-primary btn-block\\">Login</button></form></div></div>"`;
